src/main/java/Nova.java

```diff
@@ -0,0 +1,214 @@
+                    Task.getNumberOfTasks() + ".");
+            return -1;
+        }
+
+        return taskIndex;
+    }
+
+    private static void markTask(Task[] tasks, int taskIndex) {
+        tasks[taskIndex - 1].markDone();
+        displayMessage("Nice! I've marked this task as done:" + NEW_LINE + "  " +
+                tasks[taskIndex - 1].getTaskInfo());
+    }
+
+    private static void unmarkTask(Task[] tasks, int taskIndex) {
+        tasks[taskIndex - 1].unmarkDone();
+        displayMessage("OK, I've marked this task as not done yet:" + NEW_LINE + "  " +
+                tasks[taskIndex - 1].getTaskInfo());
+    }
+
+    private static void handleTodoCase(String[] words, Task[] tasks) {
+        if (words.length != 2) {
+            displayMessage("Invalid input for 'todo'." + NEW_LINE + "Usage: todo <task description>");
+            return;
+        }
+        tasks[Task.getNumberOfTasks()] = new Todo(words[1]);
+    }
+
+    private static void handleDeadlineCase(String[] words, Task[] tasks) {
+        words = validateDeadlineInput(words);
+        if (words == null) {
+            return;
+        }
+        tasks[Task.getNumberOfTasks()] = new Deadline(words[0], words[1]);
+    }
+
+    private static String[] validateDeadlineInput(String[] input) {
+        String[] splitInput = new String[2];
+        final String errorMessage = "Invalid input for 'deadline'." + NEW_LINE +
+                "Usage: deadline <task description> /by <due date>";
+        if (input.length != 2) {
+            displayMessage(errorMessage);
+            return null;
+        }
+        input = input[1].split(" /by ");
+        if (input.length != 2) {
+            displayMessage(errorMessage);
+            return null;
+        }
+        splitInput[0] = input[0];
+        splitInput[1] = input[1];
+        return splitInput;
+    }
+
+    private static void handleEventCase(String[] words, Task[] tasks) {
+        words = validateEventInput(words);
+        if (words == null) {
+            return;
+        }
+        tasks[Task.getNumberOfTasks()] = new Event(words[0], words[1], words[2]);
+    }
+
+    private static String[] validateEventInput(String[] input) {
+        String[] splitInput = new String[3];
+        final String errorMessage = "Invalid input for 'event'." + NEW_LINE +
+                "Usage: event <task description> /from <start time> /to <end time>";
+        if (input.length != 2) {
+            displayMessage(errorMessage);
+            return null;
+        }
+        input = input[1].split(" /from ");
+        if (input.length != 2) {
+            displayMessage(errorMessage);
+            return null;
+        }
+        splitInput[0] = input[0];
+        input = input[1].split(" /to ");
+        if (input.length != 2) {
+            displayMessage(errorMessage);
+            return null;
+        }
+        splitInput[1] = input[0];
+        splitInput[2] = input[1];
+        return splitInput;
+    }
+}
```

Comment:
Coding quality is generally good, use of proper indentation and naming conventions.